src/indexing/EventIndex.js

```diff
@@ -507,7 +507,14 @@ export default class EventIndex extends EventEmitter {
             try {
                 for (let i = 0; i < redactionEvents.length; i++) {
                     const ev = redactionEvents[i];
-                    await indexManager.deleteEvent(ev.getAssociatedId());
+                    const eventId = ev.getAssociatedId();
+
+                    if (eventId) {
+                        await indexManager.deleteEvent(eventId);
+                    } else {
+                        console.log("EventIndex: Redaction event doesn't contain a",
+                                    "valid associated event id", ev);
```

Comment:
We should probably use `warn`, also surely this fits within one line?
```suggestion
                        console.warn("EventIndex: Redaction event doesn't contain a valid associated event id", ev);
```